app/src/main/java/org/mozilla/fenix/onboarding/view/Onboarding.kt

```diff
@@ -0,0 +1,277 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Enum that represents the onboarding screen that is displayed.
+ */
+private enum class OnboardingState {
+    Welcome,
+    SyncSignIn
+}
+
+/**
+ * A screen for displaying a welcome and sync sign in onboarding.
+ *
+ * @param isSyncSignIn Whether or not the user is signed into their Firefox Sync account.
+ * @param onDismiss Invoked when the user clicks on the close or "Skip" button.
+ * @param onSignInButtonClick Invoked when the user clicks on the "Sign In" button
+ */
+@Composable
+fun Onboarding(
+    isSyncSignIn: Boolean,
+    onDismiss: () -> Unit,
+    onSignInButtonClick: () -> Unit,
+) {
+    var onboardingState by remember { mutableStateOf(OnboardingState.Welcome) }
+
+    Column(
+        modifier = Modifier
+            .background(FirefoxTheme.colors.layer1)
+            .fillMaxSize()
+            .padding(horizontal = 16.dp, vertical = 32.dp)
```

Comment:
The top 32dp padding is a bit arbitrary here because ideally I would specify [statusBarsPadding()](https://developer.android.com/reference/kotlin/androidx/compose/ui/Modifier#(androidx.compose.ui.Modifier).statusBarsPadding()), but we also need a Jetpack Compose version bump for this and https://google.github.io/accompanist/insets/ is deprecated since it is incorporated in the Compose Foundation library now. However, the bottom vertical should match the designs.

Reply:
A-C has a method just for this - [getStatusBarHeight](https://github.com/mozilla-mobile/android-components/blob/612514618c42466f1b4b2c5da3b5be9ba3193de4/components/support/utils/src/main/java/mozilla/components/support/utils/StatusBarUtils.kt#L18) but it is slightly async in nature.

We could also use https://github.com/mozilla-mobile/fenix/blob/77b959ec979cd806cd64acf817263725b6a6fec1/app/src/main/java/org/mozilla/fenix/ext/View.kt#L163-L164 to get the value synchronously and pass it from `HomeOnboardingDialogFragment` to `Onboarding`. 
This would ensure support for all configurations out there.

Reply:
Ended up adding https://google.github.io/accompanist/insets/ and will remove it once we bump the Jetpack Compose version.